### PR TITLE
Fix plugin /public handling for subpaths

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Plugins support a `public` folder in the bundle automatically being accessible at `<site_url>/plugins/<plugin_id/public/`, but it seems support for a `site_url` with a subpath has been broken for some time.

I tried to figure out when this stopped working, but gave up after a while and just focused on the requisite changes plus tests.

#### Ticket Link
None.

#### Release Note
```release-note
Fix plugin /public handling for subpaths
```

### CODE REVIEW

**Review by mickmister — APPROVED** [review]

**Comment by lieut-data** [comment]
@mickmister, apologies for the churn, but heads up I just pushed a change to /reduce/ the effective delta in this file and make it clearer what's happening where. (I also added more testing coverage to boot.) No semantic changes, but I'm always hesitant touching code this deep :/

**Review by mickmister — APPROVED** [review]

File: app/plugin_requests.go [hunk 1]
```diff
@@ -111,8 +110,13 @@ func (ch *Channels) ServePluginPublicRequest(w http.ResponseWriter, r *http.Requ
...
 	}
 
+	subpath, err := utils.GetSubpathFromConfig(ch.cfgSvc.Config())
+	if err != nil {
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+	}
+
 	publicFilePath := path.Clean(r.URL.Path)
-	prefix := fmt.Sprintf("/plugins/%s/public/", pluginID)
+	prefix := path.Join(subpath, "plugins", pluginID, "public")
```
Comment (hanzei) [inline app/plugin_requests.go:119]: Should [`net/url.JoinPath`](https://pkg.go.dev/net/url@go1.19.4#JoinPath) be used instead of `path.Join` here?

**Review by hanzei — APPROVED** [review]
Good work, Jesse :+1: These are solid tests.

Just one suggestion below.

**Comment by hanzei** [comment]
Merged `master` to (hopefully)  fix CI

File: app/plugin_requests.go (hunk 1 above)
Comment (lieut-data) [reply app/plugin_requests.go:119]: Alas, I think the server remains on Go 1.18, precluding the use of this improved API.

**Comment by amyblais** [comment]
@mkraft / @stylianosrigas Please help prioritize this PR review.

**Comment by amyblais** [comment]
@DHaussermann Please help prioritize this PR review.

File: app/plugin_requests_test.go [hunk 2]
```diff
@@ -4,23 +4,67 @@
...
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/utils/fileutils"
 )
 
 func TestServePluginPublicRequest(t *testing.T) {
+	t.Skip()
```
Comment (hanzei) [inline app/plugin_requests_test.go:23]: @lieut-data Just to double check: Are you planing to enable the test in this PR?

Comment (lieut-data) [reply app/plugin_requests_test.go:23]: Thanks for checking, @hanzei! I'm trying to figure out why running this test leaves the system in a state that breaks other tests. Definitely don't want to leave it disabled. This has also been broken for some time, so I'd be fine with deprioritizing for v7.7.

Comment (lieut-data) [reply app/plugin_requests_test.go]: @hanzei, fixed! Was the plugin clean up issue as I expected, I just needed to tweak the approach.

**Comment by DHaussermann** [comment]
@amyblais and @lieut-data I'm not certain how I could validate this. I can do some regression testing but, I don't know of a way to deploy this to an instance running on a subpath. This typically only happens once we reach the RC phase of release. 

@lieut-data, @hanzei or @mickmister any thoughts to share here about the scope or risk of this change? Maybe it's suitable to test this post-merge?

**Comment by DHaussermann** [comment]
Tested and passed 
- I have done some regression testing with GitLab plugin (though other plugins are also affected)
- With both the current release of GitLab as well as the build here for subpath support https://github.com/mattermost/mattermost-plugin-gitlab/pull/347 No issue were found. 
- LGTM!

@lieut-data as I believe the release branch for `v7.7.0` already exists, we should be able to merge this to `master` now. 
Once we have an RC for `v7.8` applied to a test server I can test further to validate this solution.

**Review by DHaussermann — APPROVED** [review]

**Pull request merged**